x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/AsyncStatusResponse.java

```diff
@@ -145,9 +150,13 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVInt(successfulShards);
         out.writeVInt(skippedShards);
         out.writeVInt(failedShards);
-        if (isRunning == false) {
+        if (isRunning == false) { /// MP: Hmm, is this going to be a problem? Why is this not writeOptionalString?
             RestStatus.writeTo(out, completionStatus);
```

Comment:
Having a variably written (meaning sometimes yes, sometimes no) field here now in the middle of the serialization seems problematic. I'm not sure why this was done this way instead of using `writeOptionalString`.

Is this a concern now?
Should I have change that to a writeOptionalString string in my `TransportVersion.V_8_500_008` if clause for safety?

In that case the code would be:

```
        if (out.getTransportVersion().before(TransportVersion.V_8_500_008)) {
            if (isRunning == false) { 
                RestStatus.writeTo(out, completionStatus);
            }
        } else {
            if (isRunning == false && completionStatus != null) {
                out.writeOptionalString(completionStatus.name());
            } else {
                out.writeOptionalString(null);
            }
            // optional since only CCS uses is; it is null for local-only searches
            out.writeOptionalWriteable(clusters);
        }
```

Reply:
writeOptionalBoolean writes a fixed boolean that signals whether something else follows or not. In this case, isRunning is serialized up above and drives whether the completion status is written or not. I see how this may not be super readable, but I don't think it causes issues, technically, unless you see that the condition for writing the completion status should change to not depend on isRunning alone.

Reply:
OK. Since `isRunning=false` is a terminal state, I guess this is safe. It seems odd to me however. The "optional" part of the serialization mechanism allows for nulls, so why not encode the null in the Writeable input/output when isRunning=true? The underlying mechanism already handles this case, so adding ad-hoc logic on top of it seems redundant and confusing (like there is some reason it can't be encoded as a null in the serialization system).

Since it is already there, I'm fine with leaving it if that's the path of least resistance. I just wanted to see if you though this is something that should be changed.

Reply:
I see what you mean, yet the boolean is already serialized, only a few lines above, that is my line of thinking.